### DISCUSSION TITLE
Is there any method to load another vad model?

### DISCUSSION THREAD

**Discussion by at-src** [SYSTRAN/faster-whisper#909] [Q&A] [answered]
I am looking to integrate pyannote/segmentation for vad, is there any simple method to integrate it with faster whisper or do i need to integrate it by modifying source codes?

**Comment by Purfview** [comment, marked as answer]
You can pass the external VAD timestamps with `clip_timestamps`.

**Reply by at-src** [reply]
Tried it, its making faster whisper 2x slower. I think its running vad to produce output first only then it is performing transcription with timestamp. Does it behave the same with silero or both transcription and timestamp generation takes place simultaneously hence speeding up due to silero being integrated internally?

**Reply by Purfview** [reply]
Not sure what you meant in your post, but yes pyannote/segmentation is much slower than silero.

**Reply by at-src** [reply]
I mean if I use silero with clip_timeatamps will it take same amount of time as using faster whisper with silero vad active? Or is clip_timestamps adding some time overhead?

**Reply by Purfview** [reply]
> I mean if I use silero with clip_timeatamps will it take same amount of time as using faster whisper with silero vad active?

Yes, it should take same time.
